dagql/idtui/frontend.go

```diff
@@ -338,41 +372,47 @@ func (r renderer) renderDuration(out *termenv.Output, span *Span) {
 // 	return nil
 // }
 
-type spanFilter struct {
-	db               *DB
-	gcThreshold      time.Duration
-	tooFastThreshold time.Duration
-}
+const (
+	ShowCompletedVerbosity    = 1
+	ShowInternalVerbosity     = 2
+	ShowEncapsulatedVerbosity = 2
+	ShowSpammyVerbosity       = 3
+	ShowDigestsVerbosity      = 3
 
-func (sf spanFilter) shouldShow(opts FrontendOpts, row *TraceRow) bool {
+	DefaultVerbosity = ShowCompletedVerbosity
+)
+
+func (opts FrontendOpts) ShouldShow(tree *TraceTree) bool {
 	if opts.Debug {
 		// debug reveals all
 		return true
 	}
-	span := row.Span
-	if span.IsInternal() && opts.Verbosity < 2 {
+	span := tree.Span
+	if span.IsInternal() && opts.Verbosity < ShowInternalVerbosity {
 		// internal steps are hidden by default
 		return false
 	}
-	if row.Parent != nil && (span.Encapsulated || row.Parent.Span.Encapsulate) && row.Parent.Span.Err() == nil && opts.Verbosity < 2 {
+	if tree.Parent != nil && (span.Encapsulated || tree.Parent.Span.Encapsulate) && tree.Parent.Span.Err() == nil && opts.Verbosity < ShowEncapsulatedVerbosity {
 		// encapsulated steps are hidden (even on error) unless their parent errors
 		return false
 	}
 	if span.Err() != nil {
 		// show errors
 		return true
 	}
-	if sf.tooFastThreshold > 0 && span.Duration() < sf.tooFastThreshold && opts.Verbosity < 3 {
-		// ignore fast leaf steps; signal:noise is too poor
-		if row.Parent != nil && row.Parent.Span.SpanContext().SpanID() != sf.db.PrimarySpan {
-			return false
-		}
+	if opts.TooFastThreshold > 0 && span.Duration() < opts.TooFastThreshold && opts.Verbosity < ShowSpammyVerbosity {
+		// TODO(vito): bring this back
+		// if tree.Parent != nil && tree.Parent.Span.SpanContext().SpanID() != opts.PrimarySpan {
+		// 	return false
+		// }
+		// ignore fast steps; signal:noise is too poor
+		return false
 	}
-	if row.IsRunning {
+	if tree.IsRunningOrChildRunning {
 		// show running steps
 		return true
 	}
-	if sf.gcThreshold > 0 && time.Since(span.EndTime()) > sf.gcThreshold && opts.Verbosity < 1 {
+	if opts.GCThreshold > 0 && time.Since(span.EndTime()) > opts.GCThreshold && opts.Verbosity < ShowCompletedVerbosity {
```

Comment:
:question: this doesn't seem to be working on `tty` progress anymore?

Reply:
The verbosity level defaults to 1, so this only kicks in when you press `-`